### ISSUE TITLE
Disabled relations returning their related elements in search

### ISSUE THREAD

**Issue by billythekid** [craftcms/cms#4445]
### Description
Say I have two sections, services and people which contain entries. services has an entry field to relate people entries to the service. If I disable a people entry then search by the title of that disabled entry the service shows up. Looks like relations using search aren't checking for status of the related element during matching. Not so big a deal in the back end but in a site-search these should be filtered out as the relationship doesn't _really_ exist?

### Steps to reproduce

1. Create some entries, call one of them 'do not find me'
2. Relate a few of them by an entry field in the 'do not find me' entry.
3. Disable one of the related entries, the one named 'my disabled entry' ideally :smirk:. 
4. Clear search indexes just to make sure.
5. Search for the disabled entry `craft.entries({search: 'my disabled entry',orderBy: 'score'})`
6. Observe 'do not find me' in the results.

### Additional info

- Craft version: Craft Pro 3.1.31
- PHP version: 7.3.3
- Database driver & version: MySQL 8.0.15
- Plugins & versions:
Amazon S3 	1.2.3
Bugsnag 	2.0.4
Contact Form 	2.2.5
Contact Form Honeypot 	1.0.2
Redactor 	2.3.3.2
Retour 	3.1.16
Super Table 	2.1.20
Video Embedder 	1.1.3
XML Sitemap 	1.0.9

**Comment by brandonkelly** [comment]
I agree it’s not ideal, but then if we started factoring in the related elements’ statuses before including their titles as keywords, it would lead to two other issues:

- When one of the related elements starts off as enabled and becomes disabled, we still end up in the same situation here.
- When one of the related elements starts off as disabled and becomes enabled, its title will be missing from the source element’s keywords.

Considering those, I think it’s currently doing the correct thing by blindly including all related elements’ titles as keywords.

It is possible to change the behavior though. 

1. Untick the relational field’s “Use this field’s values as search keywords?” checkbox.
2. Install [Preparse](https://plugins.craftcms.com/preparse-field)
3. Create a new Preparse field with this template:
    ```twig
    {% for rel in element.myRelationalFieldHandle.all %}
        {% if rel.enabled %}
            {{ rel }}
        {% endif %}
    {% endfor %}
    ```
4. Assign the Preparse field to whichever field layout(s) your relational field is set to.

**Comment by billythekid** [comment]
Thanks Brandon, I've implemented this workaround and it seems to catch them all. For some reason the "Use this field's values as search keywords" option had completely escaped my brain, in combination with the preparse field value, which can be hidden, this is pretty decent.

I found I had to do some combination of `./craft resave/entries`, `./craft clear-caches/all` and update search indexes (which I know is going in 3.2) to get this to propagate through the site. Obivously new entries won't be affected as they'll parse on save before the search indexes them anyway.

Thanks again!
